### ISSUE TITLE
Add support for Scala 3

### ISSUE THREAD

**Issue by Flowdalic** [chipsalliance/chisel#3967]
Chisel currently supports Scala 2.12 and Scala 2.13. It would be great if support for Scala 3 would also be available.

**Comment by jackkoenig** [comment]
Agreed entirely, there's just a chunk of work to do. I have recently started some baby steps here: https://github.com/chipsalliance/chisel/tree/source-scala-3.

Also note that we've dropped Scala 2.12 as of 6.0.0.

**Comment by edwardcwang** [comment]
Older duplicate here #2908
